types.go

```diff
@@ -33,6 +36,110 @@ type (
 
 	// Duration representing a span of time.
 	Duration = edgedbtypes.Duration
+
+	// BoolMarshaler enables encoding user defined bool values.
+	BoolMarshaler = codecs.BoolMarshaler
+
+	// BoolUnmarshaler enables decoding user defined bool values.
+	BoolUnmarshaler = codecs.BoolUnmarshaler
+
+	// BytesMarshaler enables encoding user defined bytes values.
+	BytesMarshaler = codecs.BytesMarshaler
+
+	// BytesUnmarshaler enables decoding user defined bytes values.
+	BytesUnmarshaler = codecs.BytesUnmarshaler
```

Comment:
This long list of types clutters the documentation for the edgedb package. These types don't need to be public. User code doesn't need access to them. They just need to be documented. It is probably better either:

1. to leave them in `internal/codecs`, don't make them part of the public API, and reference them in the introduction to the edgedb package telling users that they can implement them.
2. create a second public package that only has these interfaces. Maybe `edgedbmarshalers` or something. This way it is obvious that they are part of the public API but it keeps the documentation cleaner.

Reply:
> to leave them in internal/codecs, don't make them part of the public API, and reference them in the introduction to the edgedb package telling users that they can implement them.

I'd go with this option.

Reply:
changed